geniza/annotations/views.py

```diff
@@ -30,6 +31,38 @@ class ApiAccessMixin(AccessMixin):
     raise_exception = True  # return an error instead of redirecting to login
 
 
+class AnnotationLastModifiedMixin(View):
+    """View mixin to add ETag/last modified headers."""
+
+    def get_etag(self, request, *args, **kwargs):
+        """Get etag from annotation"""
+        try:
+            anno = Annotation.objects.get(pk=kwargs.get("pk"))
+            return anno.etag
+        except Annotation.DoesNotExist:
+            return None
+
+    def get_last_modified(self, request, *args, **kwargs):
+        """Return last modified :class:`datetime.datetime`"""
+        try:
+            anno = Annotation.objects.get(pk=kwargs.get("pk"))
+            return anno.modified
+        except Annotation.DoesNotExist:
+            return None
```

Comment:
is there a way to get the annotation object once instead of twice?

Reply:
Refactored a bit based on this question and the one about the mixin—let me know if you think the changes seem reasonable.